rskj-core/src/main/java/co/rsk/core/RskImpl.java

```diff
@@ -29,12 +29,16 @@
 import org.ethereum.net.server.ChannelManager;
 import org.ethereum.net.server.PeerServer;
 import org.ethereum.vm.program.invoke.ProgramInvokeFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
+@Component
 public class RskImpl extends EthereumImpl implements Rsk {
 
     private boolean isplaying;
     private final NodeBlockProcessor nodeBlockProcessor;
 
+    @Autowired
```

Comment:
Is this autowired necessary?

Reply:
Yes, before this changes we needed to perform some initalization on `co.rsk.core.RskImpl` right after the instance was created and all this logic was part of `co.rsk.core.RskFactory`. Now all that complexity is gone and we just need to instantiate this class and inject the proper dependencies which is what accomplished here through the combination of *@Component* and *@Autowired* (in the constructor)